packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -113,42 +128,84 @@ function SelectControl< ItemType = DefaultItemType >( {
 		getItemProps,
 		selectItem,
 		selectedItem: singleSelectedItem,
+		openMenu,
+		closeMenu,
 	} = useCombobox< ItemType | null >( {
 		inputValue,
 		items: filteredItems,
+		selectedItem: null,
 		itemToString: getItemLabel,
 		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 		// @ts-ignore
-		onStateChange: ( { inputValue: value, type, selectedItem } ) => {
+		onStateChange: ( changes ) => {
+			const { inputValue: value, type, selectedItem } = changes;
 			switch ( type ) {
-				case useCombobox.stateChangeTypes.InputChange:
+				case selectControlStateChangeTypes.InputChange:
 					onInputChange( value );
 					setInputValue( value || '' );
-
 					break;
-				case useCombobox.stateChangeTypes.InputKeyDownEnter:
-				case useCombobox.stateChangeTypes.ItemClick:
-				case useCombobox.stateChangeTypes.InputBlur:
+				case selectControlStateChangeTypes.InputBlur:
+					setIsFocused( false );
+					setInputValue( value || '' );
+					break;
+				case selectControlStateChangeTypes.InputKeyDownEnter:
+				case selectControlStateChangeTypes.FunctionSelectItem:
+				case selectControlStateChangeTypes.ItemClick:
 					if ( selectedItem ) {
 						onSelect( selectedItem );
 						if ( multiple ) {
 							addSelectedItem( selectedItem );
-							setInputValue( '' );
```

Comment:
I think the accessible interaction might be to clear the input value after selection.  Otherwise it may not be obvious to non-sighted users that they are continuing to add to an existing search string. If this is a custom interaction with the new component, maybe we could just keep the text as part of its state reducer for clicked items?

Reply:
Nice catch actually, the `setInputValue` was being called further down in the case statement, but I forgot that I left the `break;` in, which would never call it, I also updated the `stateReducer` to reflect this logic.
This should now work as intended, fixed as part of: https://github.com/woocommerce/woocommerce/pull/34620/commits/ef3a13a01571a4ece613944f6af96990b6407a33